src/main/scala/cromiam/webservice/CromIamApiService.scala

```diff
@@ -44,186 +56,122 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
     Http().singleRequest(cromwellRequest)
   }
 
+  private[webservice] def authorizeThenForwardToCromwell(user: Option[String], workflowId: String, action: String, request: HttpRequest): Future[HttpResponse] = {
+    (for {
+      _ <- requestAuth(WorkflowAuthorizationRequest(user.getOrElse("anon"), workflowId, action))
+      resp <- forwardToCromwell(request)
+    } yield resp) recover {
+      case SamIamDenialException => HttpResponse(status = Unauthorized, entity = "")
+      case SamActorDown => HttpResponse(status = InternalServerError, entity = "")
```

Comment:
I thought about making it a configurable value, but the status code says all we really need to say, I think.